### PR TITLE
[bug fix] Webhook race condition

### PR DESCRIPTION
**Pull request by Pralish**

### CODE REVIEW

File: app/controllers/api/external_api_clients_controller.rb [hunk 1]
```diff
@@ -15,7 +14,11 @@
         return render json: { message: message, success: false }, status: 401 unless verified
       end
 
-      @model_definition = @external_api_client.evaluated_model_definition
+      if @external_api_client.model_definition_class_defined?
+        @model_definition = ExternalApiClient.const_get(@external_api_client.model_definition_class_name)
```
[the remark this answers is not in the document]
Comment (donrestarone) [reply app/controllers/api/external_api_clients_controller.rb:18]: @Pralish is this a false positive?

[the remark this answers is not in the document]
Comment (Pralish) [reply app/controllers/api/external_api_clients_controller.rb:18]: @donrestarone , that's correct. model_definition_class_name is not a model attribute. It's a method

**Comment by donrestarone** [comment]
# UAT 🧪 

1. once merged to RC make sure the shop works https://violet-test.net/custom-shop - checkout session 
2. and we don't get spammed when checkout session / payment is made

File: test/models/external_api_client_test.rb [hunk 2]
```diff
@@ -170,4 +170,76 @@ class ExternalApiClientTest < ActiveSupport::TestCase
...
+      ExternalApiModelExampleNew
+    "
+
+    external_api_client.update(model_definition: model_definition)
+
+    refute ExternalApiClient.const_defined?(:ExternalApiModelExample)
+    assert ExternalApiClient.const_defined?(:ExternalApiModelExampleNew)
+  end
+
+  test 'should allow save if model defination has syntax error' do
```
Comment (donrestarone) [inline test/models/external_api_client_test.rb:226]: @Pralish what is this test case exercising?

Comment (Pralish) [reply test/models/external_api_client_test.rb]: @donrestarone , Sorry about that. It's a typo. It should have been 'should not allow save if model defination has syntax error'. Fixed it.

ExternalApiClient won't be valid if model_definition has syntax error. Previously we were able to save even when there were syntax errors. And we would only find out once we ran the plugin. 

In this case, the model definition is missing a 'end'. That's why it won't be saved.

Comment (donrestarone) [reply test/models/external_api_client_test.rb]: @Pralish no worries-- I just merged the geolocation PR, please catch this up-to master and remove the unsafe label if/when CI passes

**Pull request merged**